### PR TITLE
[CCT-537]feat(st-tag-input): Create component

### PR DESCRIPTION
**Pull request by jvelaystratio**
### Documentation
- [x] Add the issue in PR description
- [x] Have changelog updated
- [x] You fill the milestone value
- [x] You add some label
- [x] Have example running in demo app
- [x] Review id on demo is the same of egeo folder

### Code
- [x] Pass Sass lint task
- [x] Have qaTags

### Tests
- [x] Have at least 70% tests coverage

### CODE REVIEW

File: src/egeo-demo/st-tag-input-demo/st-tag-input-demo.component.html [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        [tooltip]="'This is a Tag Input component tooltip'">
+    </st-tag-input>
+
+    <st-tag-input
+        class="st-form-field"
+
+        name="tag-input-reactive-required"
+        formControlName="tag-input-reactive-required"
+        [errorMessage]="errorReactiveMessage"
+
```
Comment (EvaMGC) [inline src/egeo-demo/st-tag-input-demo/st-tag-input-demo.component.html]: Remove blank line

File: src/egeo-demo/st-tag-input-demo/st-tag-input-demo.component.html [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        [placeholder]="'This is a Tag Input component help'"
+        [tooltip]="'This is a Tag Input component tooltip'">
+    </st-tag-input>
+
+    <st-tag-input
+        class="st-form-field"
+
+        name="tag-input-reactive-disabled"
+        formControlName="tag-input-reactive-disabled"
+
```
Comment (EvaMGC) [inline src/egeo-demo/st-tag-input-demo/st-tag-input-demo.component.html]: Remove blank lines

File: src/egeo-demo/st-tag-input-demo/st-tag-input-demo.component.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   // public disabled: boolean = false; TODO
+
+   constructor(private _fb: FormBuilder) {
+      this.reactiveForm = _fb.group({
+         'tag-input-reactive': [this.tags.reactive],
+         'tag-input-reactive-required': [this.tags.reactiveRequired, Validators.required],
+         'tag-input-reactive-disabled': [this.tags.reactiveDisabled]
+      });
+      this.reactiveForm.valueChanges.subscribe(res => console.log('Reactive Form', res));
+      // this.templateDrivenForm.valueChanges.subscribe(res => console.log('Template Driven Form', res));
```
Comment (EvaMGC) [inline src/egeo-demo/st-tag-input-demo/st-tag-input-demo.component.ts]: Remove commented line

File: src/lib/st-tag-input/st-tag-input.component.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+   NG_VALUE_ACCESSOR,
+   Validator
+} from '@angular/forms';
+
+import { Subscription } from 'rxjs/Subscription';
+
+/**
+ * @description {Component} Tag Input
+ *
+ * This components...
```
Comment (EvaMGC) [inline src/lib/st-tag-input/st-tag-input.component.ts]: Add a brief description

File: src/lib/st-tag-input/st-tag-input.component.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      { provide: NG_VALIDATORS, useExisting: forwardRef(() => StTagInputComponent), multi: true }],
+   changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class StTagInputComponent implements ControlValueAccessor, Validator, OnInit {
+
+   @Input() label: string | null = null;
+   @Input() tooltip: string | null = null;
+   @Input() placeholder: string | null = null; // TODO
+   @Input() errorMessage: string | null = null;
+
```
Comment (EvaMGC) [inline src/lib/st-tag-input/st-tag-input.component.ts]: Add description comment to inputs/outputs

File: src/lib/st-tag-input/st-tag-input.component.ts [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+   }
+
+   writeValue(data: any): void {
+      if (data && Array.isArray(data) && data.length) {
+         this.items = [];
+         for (const value of data) {
+            this.items.push(value);
+         }
+         this.onChange(this.items);
+         this._cd.detectChanges();
```
Comment (ppenalver) [inline src/lib/st-tag-input/st-tag-input.component.ts]: use markForCheck ( detectChange its for other strategies)

**Pull request merged**